### PR TITLE
Planning poker game

### PR DESCRIPTION
**Pull request by mbuhot**
Relates to Issue #15 

This PR introduces the Planning Poker game.

Planning Poker integrates with Playdate through the `GameProvider` behaviour, exposing a `RootComponent` and `slug`.

The game logic itself is expressed as an Ash API. 

This PR does not include the UI. UI will be in the next PR.

### CODE REVIEW

File: lib/playdate/games/planning_poker.ex [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  use Ash.Api, otp_app: :playdate
+  require Ash.CodeInterface
+  alias Playdate.Games.PlanningPoker
+
+  resources do
+    registry PlanningPoker.Registry
+  end
+
+  Ash.CodeInterface.define_interface(PlanningPoker, PlanningPoker.Game)
```
Comment (mbuhot) [inline lib/playdate/games/planning_poker.ex:14]: @jimsynz Do you have a preference for defining the code interface on the API module vs the individual resource modules?

File: mix.lock [hunk 2]
```diff
@@ -1,7 +1,7 @@
...
+  "ash": {:hex, :ash, "1.53.2", "d50e0e753c8a69a4a481c250d7da34cb16f05362c58de5642db20b34d5a29c42", [:mix], [{:comparable, "~> 1.0", [hex: :comparable, repo: "hexpm", optional: false]}, {:decimal, "~> 2.0", [hex: :decimal, repo: "hexpm", optional: false]}, {:earmark, "~> 1.4", [hex: :earmark, repo: "hexpm", optional: true]}, {:ecto, "~> 3.7", [hex: :ecto, repo: "hexpm", optional: false]}, {:ets, "~> 0.8.0", [hex: :ets, repo: "hexpm", optional: false]}, {:jason, ">= 1.0.0", [hex: :jason, repo: "hexpm", optional: false]}, {:nimble_options, "~> 0.3.5", [hex: :nimble_options, repo: "hexpm", optional: false]}, {:picosat_elixir, "~> 0.2", [hex: :picosat_elixir, repo: "hexpm", optional: false]}, {:sourceror, "~> 0.9", [hex: :sourceror, repo: "hexpm", optional: false]}, {:stream_data, "~> 0.5.0", [hex: :stream_data, repo: "hexpm", optional: false]}, {:timex, ">= 3.0.0", [hex: :timex, repo: "hexpm", optional: false]}], "hexpm", "a19e7080afaa1dcaef63acfab9a6dc20d040e4a6e62bdf06e58e0b6adf41dd71"},
```
Comment (mbuhot) [inline mix.lock:2]: I bumped the dependency versions thinking that we were on the `main` branch of Ash upstream.
Ended up just getting the latest patch versions of a few dependencies, so I kept the update anyway.

File: priv/repo/migrations/20220822031531_drop_version_on_votes.exs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  Updates resources based on their most recent snapshots.
+
+  This file was autogenerated with `mix ash_postgres.generate_migrations`
+  """
+
+  use Ecto.Migration
+
+  def up do
+    alter table(:planning_poker_player_votes) do
+      remove :version
```
Comment (mbuhot) [inline priv/repo/migrations/20220822031531_drop_version_on_votes.exs:12]: I realised that optimistic concurrency control with a `version` field doesn't really apply when the main way votes are recorded is through an `upsert`.

File: lib/playdate/games/planning_poker.ex (hunk 1 above)
Comment (jimsynz) [reply lib/playdate/games/planning_poker.ex:14]: At this point you know more about code interfaces than I do. Maybe @zachdaniel has some reckons?

File: mix.lock (hunk 2 above)
Comment (jimsynz) [reply mix.lock:2]: 🤷

**Review by jimsynz — APPROVED** [review]
nicely done

**Comment by zachdaniel** [comment]
I typically do the code interfaces on the resource using the `define_for` option, but I think a good argument could be made for doing it on the api. You just have to ensure you name your interfaces well enough when doing that.

**Comment by zachdaniel** [comment]
Actually...there is a reason not to do it on the Api, which is inducing compile time dependencies from the API to all of the resources. That will have pretty significant effects on compile times as things grow. You might want to put the interface in a different module.

**Comment by zachdaniel** [comment]
Very possibly a premature optimization though, easily addressed if it becomes a problem.

**Pull request merged**